pkg/signup/signup_service.go

```diff
@@ -240,25 +242,47 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 }
 
 // GetUserSignup is used to return the actual UserSignup resource instance, rather than the Signup DTO
-func (s *ServiceImpl) GetUserSignup(userID string) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) GetUserSignup(userID string) (*v1alpha1.UserSignup, error) {
 	// Retrieve UserSignup resource from the host cluster
 	userSignup, err := s.UserSignups.Get(userID)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil, nil
-		}
 		return nil, err
 	}
 
 	return userSignup, nil
 }
 
 // UpdateUserSignup is used to update the provided UserSignup resource, and returning the updated resource
-func (s *ServiceImpl) UpdateUserSignup(userSignup *crtapi.UserSignup) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) UpdateUserSignup(userSignup *v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
 	userSignup, err := s.UserSignups.Update(userSignup)
 	if err != nil {
 		return nil, err
 	}
 
 	return userSignup, nil
 }
+
+// CheckPhoneNumberValid checks if the phone number has been banned. If so, return
+// an internal server error. If not, check if a signup with a different userID
+// exists. If so, return an internal server error. Otherwise, return without error.
+func (s *ServiceImpl) CheckPhoneNumberValid(userID, countryCode, phoneNumber string) error {
```

Comment:
Maybe something like `PhoneNumberAlreadyInUse()` would be a better name?

Reply:
sure, updated as part of: a7be0d46781f1feea1afdfca64e727a883b05d47